src/protocol/context/malicious.rs

```diff
@@ -72,6 +74,23 @@ impl<'a, F: Field> MaliciousContext<'a, F> {
         self.inner.upgrade(record_id, input, zeros_at).await
     }
 
+    /// Upgrade an input vector using this context.
+    /// # Errors
+    /// When the multiplication fails. This does not include additive attacks
+    /// by other helpers.  These are caught later.
+    pub async fn upgrade_vector<SS: Substep>(
+        &self,
+        step: &SS,
+        input: Vec<Replicated<F>>,
```

Comment:
```suggestion
        input: impl IntoIter<Item = Replicated<F>>,
```
Or the equivalent with generic parameters.  I wonder if we'll regret not tackling the streaming problem sooner...